src/ScriptCs.Hosting/Package/PackageObject.cs

```diff
@@ -60,6 +62,28 @@ public IEnumerable<string> GetCompatibleDlls(FrameworkName frameworkName)
             IEnumerable<IPackageFile> compatibleFiles;
             VersionUtility.TryGetCompatibleItems(frameworkName, dlls, out compatibleFiles);
 
+            // HACK: Delete unnecessary  temporary NuGet files
+            List<string> tempDirectories = new List<string>();
+            foreach (PhysicalPackageFile file in dlls)
+            {
+                var match = Regex.Match(((PhysicalPackageFile)dlls.First()).SourcePath, "(" + Path.Combine(Path.GetTempPath(), "nuget").Replace("\\", "\\\\") + "\\\\[^\\\\]+?)\\\\");
+                if (match.Success && match.Groups.Count > 1)
+                {
+                    tempDirectories.Add(match.Groups[1].Value); 
+                }
+
+            }
+
+            foreach (string directory in tempDirectories.Distinct())
+            {
+                if (Directory.Exists(directory))
+                {
+                    Directory.Delete(directory, true);
+                }
+            }
+
+            // /HACK
+
```

Comment:
Wow, that is *ugly!* :astonished: 

In that case I agree, this is the right place to put this hack.